oauth2/oauth2-client/src/main/java/org/springframework/security/oauth2/client/web/reactive/function/client/ServerOAuth2AuthorizedClientExchangeFilterFunction.java

```diff
@@ -430,15 +503,231 @@ private UnAuthenticatedReactiveOAuth2AuthorizedClientManager(
+		 * @param principal the principle to authorize
+		 * @return a {@link Mono} that emits the authorized client after the authorization attempt succeeds
+		 *         and the {@link #authorizationSuccessHandler} has completed,
+		 *         or completes with an exception after the authorization attempt fails
+		 *         and the {@link #authorizationFailureHandler} has completed
+		 */
+		private Mono<OAuth2AuthorizedClient> authorizeAndNotify(
+				OAuth2AuthorizationContext authorizationContext,
+				Authentication principal) {
+
+			return this.authorizedClientProvider.authorize(authorizationContext)
+					.flatMap(authorizedClient -> notifyAuthorizationSuccess(authorizedClient, principal))
+					.onErrorResume(OAuth2AuthorizationException.class, e -> notifyAuthorizationFailure(e, principal));
+		}
+
+		/**
+		 * Notifies the {@link #authorizationSuccessHandler} of the successful authorization.
+		 *
+		 * @param authorizedClient the successfully authorized client
+		 * @param principal the principal associated with the successfully authorized client
+		 * @return a {@link Mono} that emits the authorized client after the {@link #authorizationSuccessHandler} completes.
+		 */
+		private Mono<OAuth2AuthorizedClient> notifyAuthorizationSuccess(
+				OAuth2AuthorizedClient authorizedClient,
+				Authentication principal) {
+
+			return authorizationSuccessHandler.onAuthorizationSuccess(new OAuth2AuthorizationSuccessEvent(
+							authorizedClient,
+							principal))
+					.thenReturn(authorizedClient);
+		}
+
+		/**
+		 * Notifies the {@link #authorizationFailureHandler} of the failed authorization.
+		 *
+		 * @param authorizationException the failure that occurred
+		 * @param principal the principal associated with the failed authorization attempt
+		 * @return a {@link Mono} that completes with the given authorization exception after the {@link #authorizationFailureHandler} completes.
+		 */
+		private Mono<OAuth2AuthorizedClient> notifyAuthorizationFailure(
+				OAuth2AuthorizationException authorizationException,
+				Authentication principal) {
+			return authorizationFailureHandler.onAuthorizationFailure(new OAuth2AuthorizationFailureEvent(
+							authorizationException,
+							principal))
+					.then(Mono.error(authorizationException));
+		}
+
+		/**
+		 * Gets the handler that handles successful authorizations.
+		 * @return the handler that handles successful authorizations.
+		 * @see DefaultReactiveOAuth2AuthorizedClientManager.DefaultAuthorizationSuccessHandler
+		 */
+		public ReactiveOAuth2AuthorizationSuccessHandler getAuthorizationSuccessHandler() {
+			return authorizationSuccessHandler;
+		}
+
+		/**
+		 * Gets the handler that handles authorization failures.
+		 * @return the handler that handles authorization failures.
+		 * @see DefaultReactiveOAuth2AuthorizedClientManager.DefaultAuthorizationFailureHandler
+		 */
+		public ReactiveOAuth2AuthorizationFailureHandler getAuthorizationFailureHandler() {
```

Comment:
Please remove as per previous comment. As well, this class is `private`.

Reply:
Removed